Source/ARTNSError+ARTUtils.m

```diff
@@ -0,0 +1,28 @@
+//
+//  NSError+ARTUtils.m
+//  Ably
+//
+//  Created by Łukasz Szyszkowski on 05/07/2021.
+//  Copyright © 2021 Ably. All rights reserved.
+//
+
+#import "ARTNSError+ARTUtils.h"
+
+@implementation NSError (ARTUtils)
+
+- (NSError *)errorWithRequestId:(NSString *)requestId {
+    if (requestId == nil) {
+        return self;
+    }
+    
```

Comment:
This should be a static method which always return a new instance.

Reply:
not exactly, I need a previous configuration of `NSError`, like `userInfo`, `code` etc. If I'll make it static I should add another argument (`NSError`) then. Since `NSError` can't be simply mutated I can't add `request_id` key: value to `userInfo`. WDYT?

Reply:
I think it's still better to add one more argument. @ben-xD?

Reply:
@maratal Ben is not able to do obj-c review anymore. Actually, only you and @QuintinWillison can review PRs.

Reply:
https://github.com/ably/ably-cocoa/commit/8a2fc0ed5c01ef2223cd75fddef8416f4f84ac87